controller/volume_attachment_controller.go

```diff
@@ -289,6 +291,41 @@ func (vac *VolumeAttachmentController) reconcile(vaName string) (err error) {
 	return vac.handleVAStatusUpdate(va, vol)
 }
 
+// handleNodeDrain delete ui attachment ticket from the va when the target node is being drained
+func (vac *VolumeAttachmentController) handleNodeDrain(va *longhorn.VolumeAttachment, vol *longhorn.Volume) {
+	log := getLoggerForLHVolumeAttachment(vac.logger, va)
+
+	detachManuallyAttachedVolumesDuringDrain, err := vac.ds.GetSettingAsBool(types.SettingNameDetachManuallyAttachedVolumesDuringDrain)
+	if err != nil {
+		log.WithError(err).Warnf("Failed to get setting %v", types.SettingNameDetachManuallyAttachedVolumesDuringDrain)
+		return
+	}
+
+	if !detachManuallyAttachedVolumesDuringDrain {
+		return
+	}
+
+	var manualAttachmentTicket *longhorn.AttachmentTicket
+	for _, attachmentTicket := range va.Spec.AttachmentTickets {
+		if attachmentTicket.Type == longhorn.AttacherTypeLonghornAPI {
+			manualAttachmentTicket = attachmentTicket
+			break
+		}
+	}
+
+	if manualAttachmentTicket != nil {
+		unschedulable, err := vac.ds.IsKubeNodeUnschedulable(manualAttachmentTicket.NodeID)
```

Comment:
Is it possible to watch the eviction resource events of Longhorn pods to further determine if the drain operation is running, instead of only checking if the node is unscheduled?

P.S. that node is unscheduled does not mean it is being drained.

Reply:
The goal is to detach the UI attached volume from the node when it is drained
so if we can be assured that `node.spec.unschedulable` is only set to true when the node is cordoned/drained
than we don't need to add more conditions